src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/ReferenceMessageRecep.java

```diff
@@ -0,0 +1,57 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
+
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+public class ReferenceMessageRecep extends Reference {
+
+    public static final String QUALIFIER = "MIS";
+
+    private @NonNull Long messageSequenceNumber;
+    private @NonNull RecepCode recepCode;
+
+    private String buildReferenceString() {
+        return messageSequenceNumber + " " + recepCode.name();
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+        if (StringUtils.isEmpty(getReference())) {
+            throw new EdifactValidationException(getKey() + ": Attribute qualifier is required");
+        }
+    }
+
+    @Override
+    protected String getReference() {
+        return buildReferenceString();
+    }
+
+    @Override
+    protected String getQualifier() {
+        return QUALIFIER;
+    }
+
+    @Getter
+    public enum RecepCode {
+        CP("CP", "Translation successful"),
+        CA("CA", "Translation error"),
+        CI("CI", "Translation incomplete due to a fatal error during translation");
+
+        private final String code;
```

Comment:
Code might be removed and use `RecepCode.name()` method (like you did in `buildReferenceString()`) or what might be better for readability choose more descriptive enum values and use `code` (like `TRANSLATION_ERROR.getCode()`)

Reply:
I prefer `TRANSLATION_ERROR.getCode()`. Let the enum names be self-documenting and hide the "code" which is only relevant for the EDIFACT message.

Are the descriptions actually used for anything?

Reply:
Not used, but might be helpful when debugging to understand what they mean. Ill add them as comments

Reply:
If you'll use descriptive enum names then comments might not be needed

Reply:
Can we leave those values as they are now? When debugging, you'll see the description immediately

Reply:
I'd prefer if the enum had a sensible name and we dropped the unused description property.